feat: show retry count on run details page

boxofcrates opened the pull request:
Add a 'retry count' to the run details page, showing the number of retries for each passing test, and a total count of retries for all passing tests.

**Per-spec table on run details page**

With retried tests:
<img width="967" alt="Screen Shot 2021-05-16 at 9 42 04 AM" src="https://user-images.githubusercontent.com/69311002/118399671-3ccfce00-b62c-11eb-96cc-0facb048e4b8.png">

With skipped tests:
<img width="982" alt="Screen Shot 2021-05-16 at 9 42 36 AM" src="https://user-images.githubusercontent.com/69311002/118399672-3ccfce00-b62c-11eb-9e52-d8b7203a5774.png">

With both retried and skipped tests:
<img width="965" alt="Screen Shot 2021-05-16 at 9 42 57 AM" src="https://user-images.githubusercontent.com/69311002/118399683-4ce7ad80-b62c-11eb-87b3-2a650f3257b5.png">

**Overall run summary on run details page**

With retried tests:
<img width="1018" alt="Screen Shot 2021-05-16 at 9 43 43 AM" src="https://user-images.githubusercontent.com/69311002/118399704-60931400-b62c-11eb-8d4a-b45d5370d14a.png">


Without retried tests:
<img width="961" alt="Screen Shot 2021-05-16 at 9 43 38 AM" src="https://user-images.githubusercontent.com/69311002/118399698-5bce6000-b62c-11eb-8a9c-7be21796b7d2.png">

Test updates:
<img width="637" alt="Screen Shot 2021-05-16 at 9 53 10 AM" src="https://user-images.githubusercontent.com/69311002/118399738-959f6680-b62c-11eb-8db1-7b1192adad6c.png">

--

Thanks a lot for the PR 👍🏻

Your contribution is very valuable, we **really appreciate your time and effort**!

For bug fixes:

- [ ] Add a reference to the original issue
- [ ] Add a test and / or some kind of instructions to verify the fix

---

For new features:

- [ ] It's better to first discuss features proposal in [Discussions](https://github.com/sorry-cypress/sorry-cypress/discussions) or [Slack](https://sorry-cypress.slack.com/join/shared_invite/zt-eis1h6jl-tJELaD7q9UGEhMP8WHJOaw#/)
- [ ] Describe the use-case in details
- [ ] Ensure the solution is backwards-compatible
- [ ] Test it. Submit screenshots / outputs / tests together with the PR.

--- Side note ---

Time is very valuable resource and we are trying to use it wisely - your time and our time. The reality of maintaining on open-source software is:

- Adding a new code and / or feature complicates the maintenance
- Most people disappear after contributing a feature, leaving the support to core maintainers

Given that, please don't be offended if your PR is rejected or significantly modified.

P.S.
Here's an interesting discussion about OSS contributions https://changelog.com/podcast/433

❤️

agoldis commented:
Thanks @boxofcrates, looks really good!

Do you mind elaborating on the formatting changes? Curious to learn about the motivation :)

boxofcrates commented:
@agoldis for sure:

comma-dangle rule: the codebase generally seems to use trailing commas, so I added that in as a lint rule, and then there were one or two places that had to be fixed.

`argsIgnorePattern`: I saw a few `no-unused-vars` warnings about `_` vars that I assume were intentional, so added a `varsIgnorePattern` to match them the same as the existing `argsIgnorePattern`.

dashboard config: updated the dashboard eslintrc to extend from the main project-wide one and pull in those lint rules

Let me know your thoughts, happy to back any of those changes out of this PR.

agoldis commented:
@boxofcrates thanks a lot for your contribution! Nicely done! 👏🏻
